src/eval-frame/actions/fetch-cell-parser.js

```diff
@@ -0,0 +1,109 @@
+export function isValidVarname(varName) {
+  if (varName !== varName.replace(/;/g, '')) {
+    return false
+  }
+  try {
+    eval(`var ${varName};`) // eslint-disable-line no-eval
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+export function isRelPath(path) {
+  // super dumb check -- just see if it has http or https at the front
+  return !(
+    path.toLowerCase().indexOf('http://') === 0
+    || path.toLowerCase().indexOf('https://') === 0
+  )
+}
+
+export function parseFileLine(fetchCommand) {
+  return { filePath: fetchCommand, isRelPath: isRelPath(fetchCommand) }
+}
+
+export function parseAssignmentCommand(fetchCommand) {
+  const varName = fetchCommand
+    .substring(0, fetchCommand.indexOf('=')).trim()
+  if (!isValidVarname(varName)) { return { error: 'INVALID_VARIABLE_NAME' } }
+  const filePath = fetchCommand
+    .substring(fetchCommand.indexOf('=') + 1).trim()
+  return { varName, filePath, isRelPath: isRelPath(filePath) }
+}
+
+export function commentOnlyLine(line) {
+  // if, after trimming the whitespace, the first two
+  // chars are '//', then it's a comment line
+  return line.trim().substr(0, 2) === '//'
+}
+
+export function emptyLine(line) {
+  // if, after stripping all the whitespace it's empty
+  return line.replace(/\s+/g, '') === ''
+}
+
+export function parseFetchCellLine(line) {
+  // intitial sketch of syntax at:
+  // https://github.com/iodide-project/iodide/issues/1009
+  // syntax is:
+  // ${fetchType}:${one or more whitespace}...
+  // ...${fetch command (var name, path, etc)}...
+  // ...${optional comment: zero or more spaces,
+  //     followed by ' //' (which includes a space),
+  //     followed by anything}
+  if (emptyLine(line)) return undefined
+  if (commentOnlyLine(line)) return undefined
+
+  const [fetchType, fetchContent] = line.trim().split(': ') // .map(s => s.)
+  if (fetchContent) {
+    // this switch is only entered if the line contains ': ', in which case
+    // fetchContent is defined
+
+    // first, strip out comment from the end of line (if it exists)
+    const fetchCommand = fetchContent.trim().split(' //')[0].trim()
+    switch (fetchType) {
+      case 'text':
+        return Object.assign({}, { fetchType }, parseAssignmentCommand(fetchCommand))
+      case 'json':
+        return Object.assign({}, { fetchType }, parseAssignmentCommand(fetchCommand))
+      case 'blob':
+        return Object.assign({}, { fetchType }, parseAssignmentCommand(fetchCommand))
```

Comment:
I think you could shorten this to:

```js
case 'text':
case 'json':
case 'blob':
   return Object.assign({}, { fetchType }, parseAssignmentCommand(fetchCommand))
```

Likewise for the js/css cases below.

Reply:
+1 good catch